### PR TITLE
feat: Add versions/build list to `pixi search` output

### PR DESCRIPTION
**Pull request by delsner**
# Motivation

Different to e.g. `micromamba search`, `pixi search` does not provide information about other versions/builds of the found package (xref #378).

# Changes

Add versions/builds list for `pixi search` output.

Joint work with @pavelzw.

### CODE REVIEW

**Comment by pavelzw** [comment]
image
this should be `pyhd8ed1ab_1` and not `pyhd8ed1ab_0`

**Comment by pavelzw** [comment]
<details>
<summary>pixi search polarify</summary>

image

</details>

<details>
<summary>pixi search conda-forge-pinning (long versions)</summary>

image

</details>

<details>
<summary>pixi search polars (multiple builds for newest version)</summary>

image

</details>

**Comment by ruben-arts** [comment]
Thanks for this great improvement!

This looks great already, what makes it draft for you?

**Comment by delsner** [comment]
The PR still lacks adequate testing. I want to include that before marking it as ready to review.

**Comment by wolfv** [comment]
Just FYI I also had a little stab at revamping the whole search topic here: https://github.com/prefix-dev/pixi/pull/2390 

If you want to incorporate anything from there, feel free! :)

**Comment by pavelzw** [comment]
@wolfv yes we plan to improve pixi search in a series of PRs to eventually have it in a way that we can close #378 :)

**Review by ruben-arts — CHANGES_REQUESTED** [review]
Thanks for the contribution! I like the output improvements and the idea of the list of builds. The code looks good to me.

For the tests I would like to have them less hardcoded to make it more friendly to developers touching other parts of the code and then breaking these tests.

File: src/cli/search.rs [hunk 1]
```diff
@@ -442,3 +554,70 @@ fn print_matching_packages<W: Write>(
...
+#[cfg(test)]
+mod tests {
+    use crate::cli::{
+        cli_config::{ChannelsConfig, ProjectConfig},
+        search::{execute_impl, Args},
+    };
+    use rattler_conda_types::{NamedChannelOrUrl, Platform};
+
+    #[rstest::fixture]
+    fn build_string() -> &'static str {
```
[the remark this answers is not in the document]
Comment (delsner) [reply src/cli/search.rs:567]: Thanks, I agree that using the mock databases is a lot easier to maintain. I refactored the tests to use a mock database in [c483acf](https://github.com/prefix-dev/pixi/pull/2440/commits/c483acfe230d8adbfc2d022d380269fd7a77bcbd).

File: tests/integration_rust/search_tests.rs [hunk 2]
```diff
@@ -65,3 +66,64 @@ async fn search_return_latest_across_everything() {
...
+
+    let mut out = Vec::new();
+    let builder = pixi.search("foo".to_string());
+    let result = search::execute_impl(builder.args, &mut out)
+        .await
+        .unwrap()
+        .unwrap();
+    let output = String::from_utf8(out).unwrap();
+
+    assert_eq!(result.len(), 1);
```
Comment (pavelzw) [inline tests/integration_rust/search_tests.rs:123]: do you think it makes sense to snapshot result via `insta`?

Comment (delsner) [reply tests/integration_rust/search_tests.rs:123]: I think there's no additional value provided by the snapshot. Why do you think it's helpful?

Comment (pavelzw) [reply tests/integration_rust/search_tests.rs:123]: then you would more easily see whether the whitespaces of our output were set correctly

Comment (delsner) [reply tests/integration_rust/search_tests.rs:123]: Ah wait, I thought you meant the line `assert_eq!(result.len(), 1);` which only checks for repo data records. We could move the assertions regarding `output` into a snapshot, yes.

Comment (delsner) [reply tests/integration_rust/search_tests.rs:124]: [76cb5d1](https://github.com/prefix-dev/pixi/pull/2440/commits/76cb5d1f28a8f88d50af5e493c28878240ed6cb2)

**Review by pavelzw — APPROVED** [review]
💜

**Review by ruben-arts — APPROVED** [review]
I love it! No more remarks

**Pull request merged**